src/axislike.rs

```diff
@@ -688,3 +705,105 @@ impl From<DualAxisData> for Vec2 {
         data.xy
     }
 }
+
+/// The shape of the deadzone for a [`DualAxis`] input.
+///
+/// Uses the x and y thresholds on [`DualAxis`] to set the shape diamensions
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq)]
+pub enum DeadZoneShape {
+    /// Deadzone with a cross
+    ///
+    /// Uses the x threshold as the horizontal width and y threshold as the vertical width.
+    Cross {
+        /// The width of the horizontal rectangle.
+        horizontal_width: f32,
+        /// The width of the vertical rectangle.
+        vertical_width: f32,
+    },
+    /// Deadzone with a rectangle.
+    ///
+    /// Uses the x threshold as the width and y threshold as the length.
+    Rect {
+        /// The width of the rectangle.
+        width: f32,
+        /// The height of the rectangle.
+        height: f32,
+    },
+    /// Deadzone with a circle.
+    ///
+    /// Uses the larger of the x or y threshold as the radius
+    Ellipse {
+        /// The horizontal radius
+        radius_x: f32,
+        /// The vertical radius
+        radius_y: f32,
+    },
+}
+
+impl Eq for DeadZoneShape {}
+impl std::hash::Hash for DeadZoneShape {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {}
+}
+
+impl DeadZoneShape {
+    /// 
+    pub fn get_dual_axis_data(
+        &self,
+        dual_axis: &DualAxis,
+        input_stream: &InputStreams,
+    ) -> DualAxisData {
+        match self {
+            DeadZoneShape::Cross {
```

Comment:
Cross is a little odd compared to Rect and Ellipse. Technically the implemention of `Cross` is not really a deadzone, it's more snap zone. Instead of return (0,0) if it's inside the cross, it currently snaps the value to north, south, east, or west depending on where the joystick is pointing. Should I change it to be an actual "deadzone" or keep it as is? I'm in favor of making a deadzone instead, as the current snapping doesn't really turn the joystick into a dpad, only part of it. The corners of the dpad-joystick still have full f32 precision.